WordPress/gutenberg
# Migrate standalone widgets editor to use new preferences package

## Description
Addresses some of #31965, follows #38873

Migrates the standalone widgets editor to use the new preferences package (introduced in #38873). Previously it was using `interface`, but the plan is now to deprecate the preferences APIs in interface in favor of this new package.

The main goal here is that nothing is different from a user perspective.

## Testing Instructions
1. Delete your WP localstorage data
2. Using `trunk` open the widgets editor and dismiss the welcome guide
3. Toggle some of the preferences in the More Menu to something other than the default (e.g. set Top toolbar to active)
4. Checkout this branch and rebuild
5. Reload the widgets editor
6. The welcome guide should not show and the preferences set in step 3 should still be the same (the data migration worked)
7. Change a few more preferences
8. Reload
9. The preferences should still be the same from step 7 (preference changes are still persisted)

## Screenshots 

## Types of changes
What types of changes does your code introduce?




## Checklist:
